src/tests/ftest/util/server_utils.py

```diff
@@ -50,292 +47,352 @@ class ServerFailed(Exception):
 
-    Raises:
-        ServerFailed: if there was an error attempting to send a signal to stop
-            the processes running the servers or after sending the signal if
-            there are processes stiull running.
+        self.mode = "normal"
+        try:
+            self.check_subprocess_status(self.runner._process)
+        except CommandFailure as details:
+            print("Failed to start after format: {}".format(details))
 
-    """
-    global SESSIONS    # pylint: disable=global-variable-not-assigned
-    try:
-        if setname is None:
-            for _key, val in SESSIONS.items():
-                val.send_signal(signal.SIGINT)
-                time.sleep(5)
-                if val.poll() is None:
-                    val.kill()
-                val.wait()
-        else:
-            SESSIONS[setname].send_signal(signal.SIGINT)
-            time.sleep(5)
-            if SESSIONS[setname].poll() is None:
-                SESSIONS[setname].kill()
-            SESSIONS[setname].wait()
-        print("<SERVER> server stopped")
-
-    except Exception as error:
-        print("<SERVER> Exception occurred: {0}".format(str(error)))
-        raise ServerFailed("Server didn't stop!")
-
-    if not hosts:
-        return
-
-    # Make sure the servers actually stopped.  Give them time to stop first
-    # pgrep exit status:
-    #   0 - One or more processes matched the criteria.
-    #   1 - No processes matched.
-    #   2 - Syntax error in the command line.
-    #   3 - Fatal error: out of memory etc.
-    time.sleep(5)
-    result = pcmd(
-        hosts, "pgrep '(daos_server|daos_io_server)'", False, expect_rc=1)
-    if len(result) > 1 or 1 not in result:
-        bad_hosts = [
-            node for key in result if key != 1 for node in list(result[key])]
-        kill_server(bad_hosts)
-        raise ServerFailed(
-            "DAOS server processes detected after attempted stop on {}".format(
-                ", ".join([str(result[key]) for key in result if key != 1])))
-
-    # we can also have orphaned ssh processes that started an orted on a
-    # remote node but never get cleaned up when that remote node spontaneiously
-    # reboots
-    subprocess.call(["pkill", "^ssh$"])
-
-
-def kill_server(hosts):
-    """Forcably kill any daos server processes running on the specified hosts.
-
-    Sometimes stop doesn't get everything.  Really whack everything with this.
+        return True
 
-    Args:
-        hosts (list): list of host names where servers are running
-    """
-    kill_cmds = [
-        "pkill '(daos_server|daos_io_server)' --signal INT",
-        "sleep 5",
-        "pkill '(daos_server|daos_io_server)' --signal KILL",
-    ]
-    # Intentionally ignoring the exit status of the command
-    pcmd(hosts, "; ".join(kill_cmds), False, None, None)
+    def stop(self):
+        """Stop the server through the runner."""
+        if self.runner.job.yaml.is_nvme():
+            storage_reset(self._hosts)
```

Comment:
This is scrubbing the storage when stopping the server?  What if a post-mortem is desired involving investigating the state of the stroage when the test is done?